plugins/brain/__init__.py

```diff
@@ -1784,6 +1784,417 @@ def _get_target_view(ctx, target):
     return ctx.view
 
 
+class FindExactDuplicates(foo.Operator):
+    @property
+    def config(self):
+        return foo.OperatorConfig(
+            name="find_exact_duplicates",
+            label="Find exact duplicates",
+            light_icon="/assets/icon-light.svg",
+            dark_icon="/assets/icon-dark.svg",
+            description="Find exact duplicate media",
+            allow_delegated_execution=True,
+            allow_immediate_execution=True,
+            default_choice_to_delegated=True,
+            dynamic=True,
+        )
+
+    def resolve_input(self, ctx):
+        inputs = types.Object()
+
+        find_exact_duplicates_inputs(ctx, inputs)
+
+        notice = types.Notice(
+            label=(
+                "Executing this method will create two saved views:\n\n"
+                "`exact duplicates`: all samples whose media is an exact duplicate of one or more other samples\n\n"
+                "`representatives of exact duplicates`: one representative sample from each group of exact duplicates"
+            )
+        )
+        inputs.view("notice", notice)
+
+        view = types.View(label="Find exact duplicates")
+        return types.Property(inputs, view=view)
+
+    def execute(self, ctx):
+        response = find_exact_duplicates(ctx)
+
+        if not ctx.delegated:
+            ctx.trigger("reload_dataset")
+
+        return response
+
+    def resolve_output(self, ctx):
+        outputs = types.Object()
+        outputs.str(
+            "num_total_dups",
+            label="Number of samples with exact duplicates",
+        )
+        outputs.str(
+            "num_unique_dups",
+            label="Number of exact duplicate groupings",
+        )
+        view = types.View(label="Exact duplicate results")
+        return types.Property(outputs, view=view)
+
+
+def find_exact_duplicates_inputs(ctx, inputs):
+    get_target_view(ctx, inputs)
+
+
+def find_exact_duplicates(ctx):
+    target = ctx.params.get("target", None)
+
+    dataset = ctx.dataset
+    target_view = _get_target_view(ctx, target)
+
+    duplicates_dict = fob.compute_exact_duplicates(target_view)
+
+    rep_ids = []
+    dup_ids = []
+    for rep_id, neighbor_ids in duplicates_dict.items():
+        rep_ids.append(rep_id)
+        dup_ids.append(rep_id)
+        dup_ids.extend(neighbor_ids)
+
+    dups_view = dataset.select(dup_ids, ordered=True)
```

Comment:
This select stage is bound by the mongo 16mb stage limit, meaning this function will only work for <~1 million sample datasets

Reply:
Datasets with <1M duplicates, which is a reasonable assumption.